### ISSUE TITLE
test_stateless_cluster: OnQuery Error: Code: 1068, displayText = Cannot join handle from context's runtime, cause: panic.

### ISSUE THREAD

**Issue by BohuTANG** [datafuselabs/databend#4377]
**Summary**

https://github.com/datafuselabs/databend/runs/5481335834?check_suite_focus=true#step:3:5854
```
2022-03-09T14:12:37.009778Z ERROR databend_query::servers::mysql::writers::query_result_writer: OnQuery Error: Code: 1068, displayText = Cannot join handle from context's runtime, cause: panic.

   0: common_exception::exception_code::<impl common_exception::exception::ErrorCode>::TokioError
             at common/exception/src/exception_code.rs:36:66
   1: core::ops::function::FnOnce::call_once
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/ops/function.rs:227:5
      <core::result::Result<T,E> as common_exception::exception::ToErrorCode<T,E,CtxFn>>::map_err_to_code::{{closure}}
             at common/exception/src/exception.rs:196:13
      core::result::Result<T,E>::map_err
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/result.rs:842:27
      <core::result::Result<T,E> as common_exception::exception::ToErrorCode<T,E,CtxFn>>::map_err_to_code
             at common/exception/src/exception.rs:194:9
   2: databend_query::servers::mysql::mysql_interactive_worker::InteractiveWorkerBase<W>::exec_query::{{closure}}::{{closure}}
             at query/src/servers/mysql/mysql_interactive_worker.rs:342:28
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
   3: databend_query::servers::mysql::mysql_interactive_worker::InteractiveWorkerBase<W>::exec_query::{{closure}}
             at query/src/servers/mysql/mysql_interactive_worker.rs:308:5
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
   4: databend_query::servers::mysql::mysql_interactive_worker::InteractiveWorkerBase<W>::do_query::{{closure}}::{{closure}}
             at query/src/servers/mysql/mysql_interactive_worker.rs:286:57
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
   5: databend_query::servers::mysql::mysql_interactive_worker::InteractiveWorkerBase<W>::do_query::{{closure}}
             at query/src/servers/mysql/mysql_interactive_worker.rs:270:5
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
   6: <databend_query::servers::mysql::mysql_interactive_worker::InteractiveWorker<W> as msql_srv::AsyncMysqlShim<W>>::on_query::{{closure}}
             at query/src/servers/mysql/mysql_interactive_worker.rs:176:47
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
   7: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/future.rs:124:9
      msql_srv::AsyncMysqlIntermediary<B,S>::run::{{closure}}
             at /home/runner/.cargo/git/checkouts/msql-srv-62b5153e0120bb3a/af29f7b/src/lib.rs:1069:30
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
      msql_srv::AsyncMysqlIntermediary<B,S>::run_with_options::{{closure}}
             at /home/runner/.cargo/git/checkouts/msql-srv-62b5153e0120bb3a/af29f7b/src/lib.rs:821:17
   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
      msql_srv::AsyncMysqlIntermediary<B,S>::run_on::{{closure}}
             at /home/runner/.cargo/git/checkouts/msql-srv-62b5153e0120bb3a/af29f7b/src/lib.rs:801:66
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/future/mod.rs:91:19
      databend_query::servers::mysql::mysql_session::MySQLConnection::run_on_stream::{{closure}}::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/thread_pool/worker.rs:373:5
  14: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/thread_pool/worker.rs:352:45
      <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/blocking/task.rs:42:21
      tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/core.rs:161:17
      tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/loom/std/unsafe_cell.rs:14:9
      tokio::runtime::task::core::CoreStage<T>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/core.rs:151:13
      tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/harness.rs:467:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/panic.rs:137:14
      tokio::runtime::task::harness::poll_future
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/harness.rs:455:18
      tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/harness.rs:103:27
      tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/harness.rs:57:15
      tokio::runtime::task::raw::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/raw.rs:128:5
  15: tokio::runtime::task::raw::RawTask::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/raw.rs:80:18
      tokio::runtime::task::UnownedTask<S>::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/task/mod.rs:384:9
      tokio::runtime::blocking::pool::Task::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/blocking/pool.rs:91:9
      tokio::runtime::blocking::pool::Inner::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/blocking/pool.rs:308:17
      tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/blocking/pool.rs:288:17
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/sys_common/backtrace.rs:122:18
  16: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/thread/mod.rs:498:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/panic.rs:137:14
      std::thread::Builder::spawn_unchecked_::{{closure}}
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/thread/mod.rs:497:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/core/src/ops/function.rs:227:5
  17: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/alloc/src/boxed.rs:1854:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/alloc/src/boxed.rs:1854:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/8769f4ef2fe1efddd1f072485f97f568e7328f79/library/std/src/sys/unix/thread.rs:108:17
  18: start_thread
  19: clone

```

**Comment by BohuTANG** [comment]
Is it related to #4361 ? cc @ariesdevil

**Comment by ariesdevil** [comment]
I think this is same as #4347 that tokio runtime poll on a dropped future

**Comment by ariesdevil** [comment]
closed via #4415
